src/deepsparse/benchmark/torchscript_engine.py

```diff
@@ -0,0 +1,325 @@
+            self._model = torch.jit.load(model).eval()
+        else:
+            self._model = model
+
+        _validate_jit_model(self._model)
+
+        self._model.to(self.device)
+
+    def __call__(
+        self,
+        inp: List[numpy.ndarray],
+        val_inp: bool = True,
+    ) -> List[numpy.ndarray]:
+        """
+        Convenience function for TorchScriptEngine.run(), see @run for more details
+
+        | Example:
+        |     engine = TorchScriptEngine("path/to/.pt", batch_size=1)
+        |     inp = [numpy.random.rand(1, 3, 224, 224).astype(numpy.float32)]
+        |     out = engine(inp)
+        |     assert isinstance(out, List)
+
+        :param inp: The list of inputs to pass to the engine for inference.
+            The expected order is the inputs order as defined in the ONNX graph.
+        :param val_inp: Validate the input to the model to ensure numpy array inputs
+            are setup correctly for inference.
+        :return: The list of outputs from the model after executing over the inputs
+        """
+        return self.run(inp, val_inp)
+
+    def __repr__(self):
+        """
+        :return: Unambiguous representation of the current model instance
+        """
+        return "{}({})".format(self.__class__, self._properties_dict())
+
+    def __str__(self):
+        """
+        :return: Human readable form of the current model instance
+        """
+        formatted_props = [
+            "\t{}: {}".format(key, val) for key, val in self._properties_dict().items()
+        ]
+
+        return "{}:\n{}".format(
+            self.__class__.__qualname__,
+            "\n".join(formatted_props),
+        )
+
+    @property
+    def model_path(self) -> str:
+        """
+        :return: The local path to the model file the current instance was compiled from
+        """
+        return self._model_path
+
+    @property
+    def batch_size(self) -> int:
+        """
+        :return: The batch size of the inputs to be used with the model
+        """
+        return self._batch_size
+
+    @property
+    def device(self) -> str:
+        return self._device
+
+    @property
+    def scheduler(self) -> None:
+        """
+        :return: The kind of scheduler to execute with
+        """
+        return None
+
+    @property
+    def input_names(self) -> List[str]:
+        """
+        :return: The ordered names of the inputs.
+        """
+        return [node_arg.name for node_arg in self._eng_net.get_inputs()]
+
+    @property
+    def input_shapes(self) -> List[Tuple]:
+        """
+        :return: The ordered shapes of the inputs.
+        """
+        return [tuple(node_arg.shape) for node_arg in self._eng_net.get_inputs()]
+
+    @property
+    def output_names(self) -> List[str]:
+        """
+        :return: The ordered names of the outputs.
+        """
+        return [node_arg.name for node_arg in self._eng_net.get_outputs()]
+
+    @property
+    def output_shapes(self) -> List[Tuple]:
```

Comment:
`eng_net` doesn't exist and will cause these to fail. let's delete for now. can see what we can do for arg/shape inference later